Quadriga/src/main/java/edu/asu/spring/quadriga/web/workbench/PublicPageController.java

```diff
@@ -0,0 +1,132 @@
+            throws QuadrigaStorageException, QuadrigaAccessException {
+        ModelAndView model = new ModelAndView("auth/workbench/addpublicpage");
+        model.getModelMap().put("publicpage",
+                publicPageFactory.createPublicPageObject());
+        model.getModelMap().put("publicpageprojectid", projectid);
+        return model;
+    }
+
+    /**
+     * This method is used update the database with the information provided in
+     * the Public settings page
+     *
+     * @return json
+     */
+
+    @RequestMapping(method = RequestMethod.POST, value = "auth/workbench/{projectid}/addpublicpagesuccess")
+    public @ResponseBody ResponseEntity<String> getExistingConcepts(
+            @RequestParam("data") String data) throws QuadrigaStorageException,
+            QuadrigaAccessException {
+        if (data.isEmpty())
+            return null;
+        try {
+            return new ResponseEntity<String>("Successfully updated",
+                    HttpStatus.OK);
+        } catch (Exception e) {
+            return new ResponseEntity<String>(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    /**
+     * This method acts as a controller for handling settings for statistics
+     * page on publicly accessible page
+     *
+     * @author Ajay Modi
+     * @return model - model object
+     */
+
+    @AccessPolicies({ @ElementAccessPolicy(type = CheckedElementType.PROJECT, paramIndex = 1, userRole = {
+            RoleNames.ROLE_COLLABORATOR_ADMIN,
+            RoleNames.ROLE_PROJ_COLLABORATOR_ADMIN }) })
+    @RequestMapping(value = "auth/workbench/{projectid}/loadpublicstatistics", method = RequestMethod.GET)
+    public ModelAndView publicPageStatisticsSettingsForm(
+            @PathVariable("projectid") String projectid)
+            throws QuadrigaStorageException, QuadrigaAccessException {
+
+        ModelAndView model = new ModelAndView("auth/workbench/publicstatistics");
+        PublicStatisticsPage publicStatisticsPage = (PublicStatisticsPage) publicPageFactory
+                .createPublicStatisticsPageObject();
+
+        model.getModelMap().put("publicpageprojectid", projectid);
+        Map<String, Boolean> mapStatisticsSettings = statsSettingsService
+                .getStatisticsSettingsMap(projectid);
+        model.getModelMap().put("statistics", mapStatisticsSettings);
+        model.getModelMap().put("publicstatisticspage", publicStatisticsPage);
+        // model.getModelMap().put("statistics", listStatisticsSettings);
+        return model;
+    }
+
+    /**
+     * This method is used update the database with the information provided in
+     * the Public Statistics settings page
+     *
+     * @return json
+     */
+
+    @RequestMapping(method = RequestMethod.POST, value = "auth/workbench/{projectid}/submitpublicstatistics")
+    public @ResponseBody ResponseEntity<String> submitStatisticsSettings(
+            @ModelAttribute("SpringWeb") PublicStatisticsPage publicStatisticsPage,

```

Comment:
Yeah, should not do copy paste. Too error prone.

So, does it mean, modelandview only works with single model and view combination? Is it to represent tight bound between model and view?

In that case, just whatever pass from form would be in the formatted manner, so I can cut down all the logic from service and just pass over object to DAO to store.